source/Mlos.Python/mlos/Tracer.py

```diff
@@ -28,7 +28,7 @@ def wrapper(*args, **kwargs):
                 try:
                     result = wrapped_function(*args, **kwargs)
                     end_timestamp_ns = int(time.time() * 1000000)
-                    if end_timestamp_ns <= start_timestamp_ns:
+                    while end_timestamp_ns <= start_timestamp_ns:

```

Comment:
Curious, is there some case where time.time() goes backwards for some reason that the if statement version doesn't work?

Reply:
I've seen it go backwards a bunch of times. Never seemed to have been more than 100 but this seems more robust.